src/sdk-reference/cpp/1/protocol/add-listener/index.md

```diff
@@ -0,0 +1,45 @@
+---
+layout: sdk.html.hbs
+title: addListener
+description: Add a listener to an event
+---
+
+# addListener
+
+Adds a listener to an event.
+
+## Signature
+
+```cpp
+virtual void addListener(kuzzleio::Event event, kuzzleio::EventListener* listener) = 0;
+```
+
+## Arguments
+
+| Argument   | Type                      | Description
+| ---------- |------------------------------------------------------------------------------------------------------ | -------- |
+| `event`    | <pre>kuzzleio::Event</pre>           | An enum representing the listener [event]({{ site_base_path }}sdk-reference/essentials/event-handling)
+| `listener` | <pre>kuzzleio::EventListener*</pre> | A pointer to a c++11 lambda
+
+### **event**
+
+One of the following event:
+
+```cpp
+CONNECTED
+DISCARDED
+DISCONNECTED
+LOGIN_ATTEMPT
+NETWORK_ERROR
+OFFLINE_QUEUE_POP
+OFFLINE_QUEUE_PUSH
+QUERY_ERROR
+RECONNECTED
+JWT_EXPIRED
+ERROR
+```
+
+### **listener**
+
+A c++11 lambda which take a `const std::string`
```

Comment:
Not sure this suggestion is correct, it seems that a part of the sentence is missing.

```suggestion
A c++11 lambda which takes a `const std::string` argument
```